`tests/semantic_kernel/test_semantic_kernel_autolog.py::test_sk_invoke_agent` fails in `semantic-kernel>=1.35.1`

## How to reproduce:

```
pip install 'semantic-kernel>=1.35.1'     
pytest 'tests/semantic_kernel/test_semantic_kernel_autolog.py::test_sk_invoke_agent'
```

## Traceback:

https://github.com/mlflow/mlflow/actions/runs/16898788344/job/47873884564#step:13:55

```
mock_openai = 'http://localhost:47729'

    @pytest.mark.asyncio
    async def test_sk_invoke_agent(mock_openai):
        mlflow.semantic_kernel.autolog()
        result = await _create_and_invoke_chat_agent(mock_openai)
        assert isinstance(result, AgentResponseItem)
    
        traces = get_traces()
        assert len(traces) == 1
        trace = traces[0]
        spans = trace.data.spans
>       assert len(spans) == 3
E       AssertionError: assert 2 == 3
E        +  where 2 = len([Span(name='AutoFunctionInvocationLoop', trace_id='tr-f5142f4eae701db36baafd86b1ddb2cb', span_id='dcce8ed060111d4a', parent_id=None), Span(name='chat.completions gpt-4o-mini', trace_id='tr-f5142f4eae701db36baafd86b1ddb2cb', span_id='2e220036db1cb529', parent_id='dcce8ed060111d4a')])
```

## Potential cause

https://github.com/microsoft/semantic-kernel/pull/12834

--- a/mlflow/semantic_kernel/autolog.py
+++ b/mlflow/semantic_kernel/autolog.py
@@ -37,11 +37,10 @@ def _enable_experimental_genai_tracing():
         MODEL_DIAGNOSTICS_SETTINGS,
     )
 
-    MODEL_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics = True
-    MODEL_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics_sensitive = True
-
     AGENT_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics = True
     AGENT_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics_sensitive = True
+    MODEL_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics = True
+    MODEL_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics_sensitive = True
 
     _logger.info("Semantic Kernel Otel diagnostics is turned on for enabling tracing.")
 

--- a/mlflow/semantic_kernel/autolog.py
+++ b/mlflow/semantic_kernel/autolog.py
@@ -30,17 +30,24 @@ def _enable_experimental_genai_tracing():
     # We directly update the singleton setting object instead of using env vars,
     # because the object might be already initialized by the time we call this function.
     # https://learn.microsoft.com/en-us/semantic-kernel/concepts/enterprise-readiness/observability/telemetry-with-console
-    from semantic_kernel.utils.telemetry.agent_diagnostics.decorators import (
-        MODEL_DIAGNOSTICS_SETTINGS as AGENT_DIAGNOSTICS_SETTINGS,
-    )
+
     from semantic_kernel.utils.telemetry.model_diagnostics.decorators import (
         MODEL_DIAGNOSTICS_SETTINGS,
     )
 
-    AGENT_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics = True
-    AGENT_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics_sensitive = True
     MODEL_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics = True
     MODEL_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics_sensitive = True
+
+    try:
+        # This only exists in Semantic Kernel 1.35.1 or later.
+        from semantic_kernel.utils.telemetry.agent_diagnostics.decorators import (
+            MODEL_DIAGNOSTICS_SETTINGS as AGENT_DIAGNOSTICS_SETTINGS,
+        )
+
+        AGENT_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics = True
+        AGENT_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics_sensitive = True
+    except ImportError:
+        pass
 
     _logger.info("Semantic Kernel Otel diagnostics is turned on for enabling tracing.")
 

--- a/mlflow/semantic_kernel/autolog.py
+++ b/mlflow/semantic_kernel/autolog.py
@@ -30,7 +30,6 @@ def _enable_experimental_genai_tracing():
     # We directly update the singleton setting object instead of using env vars,
     # because the object might be already initialized by the time we call this function.
     # https://learn.microsoft.com/en-us/semantic-kernel/concepts/enterprise-readiness/observability/telemetry-with-console
-
     from semantic_kernel.utils.telemetry.model_diagnostics.decorators import (
         MODEL_DIAGNOSTICS_SETTINGS,
     )

--- a/mlflow/semantic_kernel/autolog.py
+++ b/mlflow/semantic_kernel/autolog.py
@@ -30,12 +30,18 @@ def _enable_experimental_genai_tracing():
     # We directly update the singleton setting object instead of using env vars,
     # because the object might be already initialized by the time we call this function.
     # https://learn.microsoft.com/en-us/semantic-kernel/concepts/enterprise-readiness/observability/telemetry-with-console
+    from semantic_kernel.utils.telemetry.agent_diagnostics.decorators import (
+        MODEL_DIAGNOSTICS_SETTINGS as AGENT_DIAGNOSTICS_SETTINGS,
+    )
     from semantic_kernel.utils.telemetry.model_diagnostics.decorators import (
         MODEL_DIAGNOSTICS_SETTINGS,
     )
 
     MODEL_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics = True
     MODEL_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics_sensitive = True
+
+    AGENT_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics = True
+    AGENT_DIAGNOSTICS_SETTINGS.enable_otel_diagnostics_sensitive = True
 
     _logger.info("Semantic Kernel Otel diagnostics is turned on for enabling tracing.")
 

